### PR TITLE
Upgrade to ZIO 1.0.0-RC17

### PR DESCRIPTION
**Pull request by svroonland**

### CODE REVIEW

**Comment by svroonland** [comment]
@iravid The tests seem to succeed, but fail on the closing of resources at lines like `_ <- fib.join.ignore`. Probably something to do with the fiber supervision overhaul. I can't quite figure out how to fix it.

**Comment by iravid** [comment]
Uh oh. Looks like we're not releasing today then. If you don't figure it out by tomorrow I will join in on the fun.

**Comment by svroonland** [comment]
Think I fixed everything :)

I did not process the non-mandatory changes from the release notes.

File: src/test/scala/zio/kafka/client/KafkaTestUtils.scala [hunk 1]
```diff
@@ -131,7 +131,7 @@ object KafkaTestUtils {
       ),
       250.millis,
       250.millis,
-      1
+      100
```
Comment (svroonland) [inline src/test/scala/zio/kafka/client/KafkaTestUtils.scala]: This improves test speed a bit

Comment (TimPigden) [reply src/test/scala/zio/kafka/client/KafkaTestUtils.scala]: Hi - can you explain that please. These numbers were a bit of amystery to me

Comment (svroonland) [reply src/test/scala/zio/kafka/client/KafkaTestUtils.scala]: Well, to be honest, I believe it helped to speed up a test I ran but I'd have to look closer at the exact mechanis in `RunLoop` to say more about its exact effect. 

In any case, the setting is used as the parameter to `ZStream.buffer` which:
> Allows a faster producer to progress independently of a slower consumer by bufferingup to `capacity` elements in a queue.

In backpressuring streams it helps to have some buffers to deal with non-constant processing speeds.

**Pull request merged**